### PR TITLE
feat: reservations create/get/list/delete samples

### PR DESCRIPTION
**Pull request by BigBlackWolf**
## Description

Created 4 new samples with basic operations:
- create
- delete
- get
- list

## Checklist
- [ ] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [ ] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [ ] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/reservations/reservations_test.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	"testing"
+	"time"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"cloud.google.com/go/compute/apiv1/computepb"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	"google.golang.org/protobuf/proto"
+)
+
+func createTemplate(ctx context.Context, project, templateName string) error {
```
Comment (telpirion) [inline compute/reservations/reservations_test.go:32]: issue: don't pass around context objects unless you have applied a timeout function to them.

Here and below.

File: compute/reservations/reservations_test.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+		},
+	}
+	op, err := client.Insert(ctx, req)
+	if err != nil {
+		return err
+	}
+	return op.Wait(ctx)
+}
+
+func getTemplate(ctx context.Context, project, templateName string) (*computepb.InstanceTemplate, error) {
```
Comment (telpirion) [inline compute/reservations/reservations_test.go:67]: issue: remove context object from method signature. Create a new context object in the function.

File: compute/reservations/reservations_test.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	}
+
+	req := &computepb.GetInstanceTemplateRequest{
+		Project:          project,
+		InstanceTemplate: templateName,
+	}
+	return client.Get(ctx, req)
+}
+
+func deleteTemplate(ctx context.Context, project, templateName string) error {
```
Comment (telpirion) [inline compute/reservations/reservations_test.go:80]: issue: remove context object from method signature. Create a new context object in the function.

**Review by telpirion — APPROVED** [review]
Just some minor nits around context objects in the tests. Otherwise, looks good to me!

File: compute/reservations/reservations_test.go (hunk 1 above)
Comment (BigBlackWolf) [reply compute/reservations/reservations_test.go:32]: Get it, makes sense. I see that PR was merged, but I will create separate PR and correct those things.
Thanks for the feedback!

Comment (BigBlackWolf) [reply compute/reservations/reservations_test.go:32]: Here it is https://github.com/GoogleCloudPlatform/golang-samples/pull/4371

**Pull request merged**